_automating-workflows/api/create-workflow.md

```diff
@@ -0,0 +1,247 @@
+
+#### Example Request: Register and deploy an externally hosted model (YAML)
+
+To provide a template in YAML format, specify `Content-Type: application/yaml` in the request header:
+
+```bash
+curl -XPOST "http://localhost:9200/_plugins/_flow_framework/workflow" -H 'Content-Type: application/yaml'
+```
+
+YAML templates permit comments. 
+{: .tip}
+
+The following is an example YAML template to register and deploy an externally hosted model:
+
+```yaml
+# This name is required
+name: createconnector-registerremotemodel-deploymodel
+# Other fields are optional but useful
+description: This template creates a connector to a remote model, registers it, and
+  deploys that model
+# Other templates with a similar use case can be searched
+use_case: REMOTE_MODEL_DEPLOYMENT
+version:
+  # Templates may be versioned by their authors
+  template: 1.0.0
+  # Compatibility with OpenSearch 2.12.0 and higher and 3.0.0 and higher
+  compatibility:
+  - 2.12.0
+  - 3.0.0
+# One or more workflows can be included, presently only provision is supported
+workflows:
+  provision:
+    # These nodes are the workflow steps corresponding to ML Commons APIs
+    nodes:
+    # This ID must be unique to this workflow
+    - id: create_connector_1
+      # There may be multiple steps with the same type
+      type: create_connector
+      # These inputs match the Create Connector API body
+      user_inputs:
+        name: OpenAI Chat Connector
+        description: The connector to public OpenAI model service for GPT 3.5
+        version: '1'
+        protocol: http
+        parameters:
+          endpoint: api.openai.com
+          model: gpt-3.5-turbo
+        credential:
+          openAI_key: '12345'
+        actions:
+        - action_type: predict
+          method: POST
+          url: https://${parameters.endpoint}/v1/chat/completions
+    # This ID must be unique to this workflow
+    - id: register_model_2
+      type: register_remote_model
+      # This step needs the connector_id produced as an output of the previous step
+      previous_node_inputs:
+        create_connector_1: connector_id
+      # These inputs match the Register Model API body
+      user_inputs:
+        name: openAI-gpt-3.5-turbo
+        function_name: remote
+        description: test model
+    # This ID must be unique to this workflow
+    - id: deploy_model_3
+      type: deploy_model
+      # This step needs the model_id produced as an output of the previous step
+      previous_node_inputs:
+        register_model_2: model_id
+    # Since the nodes include previous_node_inputs these are optional to define
+    # They will be added automatically and included in the stored template
+    # Additional edges may also be added here if required for sequencing
+    edges:
+    - source: create_connector_1
+      dest: register_model_2
+    - source: register_model_2
+      dest: deploy_model_3
```

Comment:
@dbwiddis do you think we should also mention that if `previous_node_inputs` are defined then defining edges is optional?

Reply:
It is mentioned immediately above in the comment (line 142-144)

Reply:
I will add a callout in the tutorial so this info is explicitly mentioned

Reply:
I've tried to mention it in multiple places, such as where previous_node_inputs are defined as well.